src/sidebar-counters/sidebar-counters.js

```diff
@@ -4,66 +4,80 @@
+}
+
+type ResultSize = $Exact<{ size: number }>
 
-    const badge = (
+export function getBadge(size: number): HTMLElement {
+    const maxReached = size > MAX_COUNTER
+    return (
         <span class="__rbb-badge">
-            <span class="__rbb-badge-counter">{size}</span>
+            <span class={`__rbb-badge-counter${maxReached ? ' -max' : ''}`}>
+                {typeof size !== 'number' ? '!' : Math.min(size, MAX_COUNTER)}
+            </span>
         </span>
     )
-    a.insertBefore(badge, a.firstChild)
-    return badge
 }
 
-export default async function addSideBarCounters() {
-    // Exit early if can't find any of the nav links
-    // that we are going to augment with badge counters
-    const branchesLink: HTMLAnchorElement = await elementReady(
-        'a[href$="branches/"]'
-    )
-    const pullRequestsLink: HTMLAnchorElement = await elementReady(
-        'a[href$="pull-requests/"]'
-    )
-
-    if (!branchesLink || !pullRequestsLink) {
-        return
+async function getCounterInfo(wantedMenuFromHref: string): Promise {
+    switch (wantedMenuFromHref) {
+        case HREF_BRANCHES:
+            return api.getBranches()
+        case HREF_PULL_REQUESTS:
+            return api.getPullrequests()
+        default:
+            return
     }
+}
 
-    const [branches, pullrequests] = await Promise.all([
-        api.getBranches(),
-        api.getPullrequests(),
-    ])
+export async function addCounterToMenuItem(menu: HTMLElement) {
+    const link: HTMLAnchorElement = menu.querySelector('a')
 
-    const branchesBadge = addBadge(branchesLink, branches)
-    const prBadge = addBadge(pullRequestsLink, pullrequests)
+    if (!link) return
 
-    const resizeButton: HTMLButtonElement = await elementReady(
-        '.ak-navigation-resize-button'
+    const hrefParts = link.getAttribute('href').split('/')
+    const wantedMenuFromHref = hrefParts.find(x => menus.includes(x))
+
+    // Exit early if can't find one of the nav links
+    // that we are going to augment with badge counters
+    if (!wantedMenuFromHref) return
+
+    const size = menusCounters[wantedMenuFromHref]
+    const badge = getBadge(size)
+
+    menu.style.overflow = 'hidden'
+    link.style.position = 'relative'
+    link.insertBefore(badge, link.firstChild)
+}
+
+export default async function addSideBarCounters() {
+    menus.forEach(async menu => {
+        const resources = await getCounterInfo(menu)
+        const { size: currentSize } = resources || {}
+        menusCounters[menu] = currentSize
+    })
+    const contentNavigationSelector =
+        'div[data-testid="Content"] div[role="presentation"]'
+    const contextualNavigationSelector =
+        'div[data-testid="ContextualNavigation"] div[role="presentation"]'
+    new SelectorObserver(
+        document.body,
+        [contentNavigationSelector, contextualNavigationSelector].join(', '),
+        function() {
+            addCounterToMenuItem(this, menusCounters)
```

Comment:
Like I said in my previous comment, we can pass the count data as an argument as you are doing here with `menuCounters` and have that variable be local to this function instead